### DISCUSSION TITLE
include of non-modular header inside framework module 'SDWebImage': '/Users/distiller/project/Tuist/Dependencies/SwiftPackageManager/.build/checkouts/SDWebImage/SDWebImage/include/SDWebImage/SDWebImageDownloaderRequestModifier.h'

### DISCUSSION THREAD

**Discussion by kusainovaka** [SDWebImage/SDWebImage#3603] [Q&A]
Circle ci get error like 
`include of non-modular header inside framework module 'SDWebImage': '/Users/distiller/project/Tuist/Dependencies/SwiftPackageManager/.build/checkouts/SDWebImage/SDWebImage/include/SDWebImage/SDWebImageDownloaderRequestModifier.h'`

To fix it AI prefer change import methods, but library keeps in cloud, local impossible to change it 
<img width="417" alt="Снимок экрана 2023-09-18 в 15 35 42" src="https://github.com/SDWebImage/SDWebImage/assets/18579305/5e185d06-5d46-41a4-a243-d160b7055bcf">
<img width="835" alt="Снимок экрана 2023-09-18 в 15 37 25" src="https://github.com/SDWebImage/SDWebImage/assets/18579305/a0223716-8f98-4e17-9d6d-eb0db872f740">

**Comment by dreampiggy** [comment]
You should provide the information in issue template.

1. SDWebImage Version
2. iOS/macOS version
3. Package Manager (Are you using bazel ? Note: We do not natively support bazel, check whether you use PodToBuild(https://github.com/bazel-xcode/PodToBUILD) or some other tools to convert CocoaPods to bazel, isn't that convert tool bugs ?)

**Reply by dreampiggy** [reply]
And, `SDWebImageDownloaderRequestModifier.h` is actually modular header (and public header).  It should be in the `module.modulemap` because cocoapods generate modulemap to `module 'SDWebImage' export *`

**Reply by dreampiggy** [reply]
For bazel if you really want to use SDWebImage, I don't think PodToBuild is the suggested way. You can even write a simple bzl mod for support bazel.

**Reply by kusainovaka** [reply]
1. SDWebImage (SPM - 5.1.0)
2. iOS version
3. no we don't use pod, only spm

**Reply by dreampiggy** [reply]
In SDWebImage 5.18.0/5.18.1, I can successfuly run `swift build` and `swift build --target SDWebImageMapKit`

![image](https://github.com/SDWebImage/SDWebImage/assets/6919743/5e474bde-cf23-40f1-a8b6-4abe3b938d35)


And also, that https://github.com/SDWebImage/SDWebImageSwiftUI using SDWebImage via SPM dependency (see that Xcodeproj's dependency), and build successfully..

Is there something configuration issue in your project 🤔️ ?
